mlir/lib/Dialect/Bufferization/Transforms/OneShotModuleBufferize.cpp

```diff
@@ -388,27 +401,33 @@ mlir::bufferization::analyzeModuleOp(ModuleOp moduleOp,
     return failure();
 
   // Analyze ops.
-  for (func::FuncOp funcOp : orderedFuncOps) {
-    if (!state.getOptions().isOpAllowed(funcOp))
+  for (FunctionOpInterface funcOp : orderedFuncOps) {
+
+    // The following analysis is specific to the FuncOp type.
+    if(!isa<FuncOp>(funcOp))
+        continue;
```

Comment:
I think this is functionally correct but could be a performance problem in practice. This piece of code is gathering information about the relationship between func args and func results. This is needed to compute aliasing values during `CallOpInterface::getAliasingValues`. For functions that were not analyzed, we assume that all CallOp operands potentially alias will all CallOp results. You may get additional buffer copies because of that. Could that become a problem?

Reply:
Yes, I can see it is bad for the performance. This part involves `FuncAnalysisState` from `FuncBufferizableOpInterfaceImpl`, which is also hard-coded with `FuncOp`. I will try replacing it with `FunctionOpInterface` as well.

Reply:
We have made all fields in `FuncAnalysisState` use `FunctionOpInterface` instead of `func::FuncOp`.